Render default apps using cluster chart (#387)
giantswarm/cluster-cloud-director: CHANGELOG.md

    on how to do this, see the [giantswarm/cluster chart readme](https://github.com/giantswarm/cluster/tree/main/helm/cluster#apps).
- Delete default-apps-cloud-director App CR.
  - ⚠️ In case you are removing the default-apps-cloud-director App CR from your gitops repo which is using Flux, and depending on
    how Flux is configured, the default-apps-cloud-director App CR may or may not get deleted from the management cluster. In case
    Flux does not delete the default-apps-cloud-director App CR from the management cluster, make sure to delete it manually.
  - App CRs (on the MC) for all default Apps will be deleted. Wait a few minutes for this to happen.
  - Chart CRs on the workload cluster will remain untouched, so all apps will continue running.
- Upgrade the cluster-cloud-director App CR to the v0.62.0 release.
  - cluster-cloud-director will deploy all default Apps, so wait a few minutes for all Apps to be successfully deployed.
  - Chart resources on the workload cluster will get updated, as newly deployed App resources will take over the reconciliation
    of the existing Chart resources.

> We're almost there, with just one more issue to fix manually.

Reviewer: ```suggestion
### Manual fixes

We're almost there, with just a couple more issues to fix manually.
```

> The VPA CRD used to be installed as an App resource from default-apps-cloud-director, but now it's being installed as a HelmRelease

Reviewer: ```suggestion
#### Vertical Pod Autoscaler

The VPA CRD used to be installed as an App resource from default-apps-cloud-director, but now it's being installed as a HelmRelease
```

1. Pause the vertical-pod-autoscaler-crd Chart CR.

Add annotation `chart-operator.giantswarm.io/paused: "true"` to the vertical-pod-autoscaler-crd Chart CR in the workload cluster:

```sh
kubectl annotate -n giantswarm chart vertical-pod-autoscaler-crd chart-operator.giantswarm.io/paused="true" --overwrite
```

2. Delete the vertical-pod-autoscaler-crd Chart CR in the workload cluster.

```shell
kubectl delete -n giantswarm chart vertical-pod-autoscaler-crd
```

Kubectl will probably hang, as the chart-operator finalizer is not removed (vertical-pod-autoscaler-crd
Chart CR has been paused). Proceed to the next step to remove the finalizer and unblock the deletion.

3. Remove finalizers from the vertical-pod-autoscaler-crd Chart CR

Open another terminal window and run the following command to remove the vertical-pod-autoscaler-crd Chart CR finalizers:

```shell
kubectl patch chart vertical-pod-autoscaler-crd -n giantswarm --type=json -p='[{"op": "remove", "path": "/metadata/finalizers"}]'
```

This will unblock the deletion and vertical-pod-autoscaler-crd will get removed, **without actually deleting the VPA CustomResourceDefinition**.

> From now on, the VPA CustomResourceDefinition will be maintained by the vertical-pod-autoscaler HelmRelease on the management cluster.

Reviewer: ```suggestion
From now on, the VPA CustomResourceDefinition will be maintained by the vertical-pod-autoscaler HelmRelease on the management cluster.

#### Observability Platform Configuration

The observability operator is responsible for creating the `<cluster-name>-observability-platform-configuration` configmap and patching the `<cluster-name>-observability-bundle` app to reference it. This configuration is lost after upgrading the cluster app. As a result, the observability operator has to be restarted.

"""shell
kubectl delete pod -n monitoring -l app.kubernetes.io/instance=observability-operator
"""
```
